### PR TITLE
fix: replace deprecated AddToScheme with Install, and deprecated Sche…

### PR DESCRIPTION
**Pull request by chengfang**
…meGroupVersion with GroupVersion.

**What type of PR is this?**

> /kind cleanup

**What does this PR do / why we need it**:
Replace deprecated functions (`SchemeGroupVersion`, `AddToScheme`) with current ones.

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:
there should be no regression introduced by this PR.

### CODE REVIEW

File: controllers/argocd/custommapper_test.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func TestReconcileArgoCD_tlsSecretMapperRepoServer(t *testing.T) {
 		resObjs := []client.Object{argocd, secret, service}
 		subresObjs := []client.Object{argocd}
 		runtimeObjs := []runtime.Object{}
-		sch := makeTestReconcilerScheme(argoproj.AddToScheme, configv1.AddToScheme, routev1.AddToScheme)
+		sch := makeTestReconcilerScheme(argoproj.AddToScheme, configv1.Install, routev1.Install)
```
Comment (svghadi) [inline controllers/argocd/custommapper_test.go:156]: Do we need to update `argoproj.AddToScheme` to `argoproj.Install` or is it that deprecation is only for specific kinds?

Comment (chengfang) [reply controllers/argocd/custommapper_test.go:156]: `argoproj.AddToScheme` appears to be a totally different function, and is not deprecated.

**Review by svghadi — APPROVED** [review]
/lgtm
Thanks @chengfang

**Pull request merged**